src/EnergyPlus/UnitarySystem.cc

```diff
@@ -8103,7 +8103,7 @@ namespace UnitarySystems {
                 this->controlCoolingSystemToSP(state, AirLoopNum, FirstHVACIteration, HXUnitOn, CompressorOn);
                 PartLoadRatio = this->m_CoolingPartLoadFrac;
                 CompressorOn = HVAC::CompressorOp::Off;
-                if (PartLoadRatio > 0.0) {
+                if ((PartLoadRatio > 0.0) || (PartLoadRatio == 0.0 && this->m_CoolingSpeedRatio > 0.0)) {
                     CompressorOn = HVAC::CompressorOp::On;
                     this->m_LastMode = CoolingMode;
                 }
```

Comment:
When there is latent load only (i.e., when there is no sensible load), then above logic `if (PartLoadRatio > 0.0)` holds true for single speed DX coil only. For variable speed DX cooling coil, the PLR can be zero when the speed ratio is greater than zero. Thus, need to add the second logic to avoid turning off the variable DX cooling coil.

Reply:
> this->m_CoolingPartLoadFrac = std::max(PartLoadFrac, SpeedRatio);

This logic may work but we will be mixing up two variables. The `m_CoolingPartLoadFrac` and `SpeedRatio` have a different meaning and resetting the `m_CoolingPartLoadFrac` that may have un-intended consequence elsewhere.

Reply:
May be use a local variable that does not affect the `m_CoolingPartLoadFrac` member variable.

Reply:
If I understand correctly, for variable speed DX coil, the `m_CoolingPartLoadFrac` value is set `0.0` whenever the `SpeedRatio` is > `0.0`. @rraustad Do you agree?